nuxeo-jsf/nuxeo-platform-contentview-jsf/src/main/java/org/nuxeo/ecm/platform/contentview/json/JSONContentViewState.java

```diff
@@ -301,7 +301,10 @@ protected static Serializable getDocumentPropertyValue(Object o) throws JSONExce
             JSONArray jsonArray = (JSONArray) o;
             ArrayList<Serializable> list = new ArrayList<Serializable>();
             for (Object aJsonArray : jsonArray) {
-                list.add(getDocumentPropertyValue(aJsonArray));
+                Serializable pValue = getDocumentPropertyValue(aJsonArray);
+                if (pValue != null && !(pValue instanceof JSONNull)) {
+                  list.add(pValue);
```

Comment:
Could you move the `instanceof JSONNull` logic down into `getDocumentPropertyValue` instead please? It makes more sense there.